presto-main/src/main/java/com/facebook/presto/sql/planner/DomainTranslator.java

```diff
@@ -689,6 +691,25 @@ protected ExtractionResult visitInPredicate(InPredicate node, Boolean complement
             InListExpression valueList = (InListExpression) node.getValueList();
             checkState(!valueList.getValues().isEmpty(), "InListExpression should never be empty");
 
+            Symbol symbol = Symbol.from(node.getValue());
+            Type type = checkedTypeLookup(symbol);
+
+            // Check if we can bypass all the processing. If all the values in the IN clause are literals,
+            // and the type is orderable, we can just directly construct the SortedRangeSet.
```

Comment:
Remove " Check if we can bypass all the processing.", it's pretty obvious that you special-case here.